File: SearchPatterns.Tests/CPP/SearchPatterns.Test/permutation_fixture.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include "lib.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace SearchPatternsTest
{		
	TEST_CLASS(PermutationFixture)
	{
	public:
		
		TEST_METHOD(FunctionMustReturnYes)
		{
			Lib::CalcPattern("use", "question");
		}

	};
}

File: SearchPatterns.Tests/CS/SearchPatterns.Test/PermutationFixture.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Permutations;

namespace SearchPatterns.Test
{
    [TestClass]
    public class PermutationFixture
    {
        [TestMethod]
        public void CalcShouldReturnYes()
        {
            Assert.AreEqual(Permutations.Lib.CalcPattern("use", "question"), "YES");
        }

        [TestMethod]
        public void CalcShouldThroughAllText()
        {
            Assert.AreEqual(Permutations.Lib.CalcPattern("use", "usgues"), "YES");
        }

        [TestMethod]
        public void CalcShouldReturnNoForUncompletePattern()
        {
            Assert.AreEqual(Permutations.Lib.CalcPattern("use", "sugestion"), "NO");
        }

        [TestMethod]
        public void CalcPatternShouldIgnoreLongPattern()
        {
            Assert.AreEqual(Permutations.Lib.CalcPattern("shortbutverylong", "short"), "NO");
        }
    }
}
